File: kmp_module/app/src/androidMain/kotlin/com/dobby/feature/logging/LoggerManagerImpl.kt
package com.dobby.feature.logging

import com.dobby.backend.GoBackendWrapper
import com.dobby.feature.logging.domain.provideGoLogFilePath

class LoggerManagerImpl(
    private val logger: Logger,
) : LoggerManager {
    override fun initLogger(): Boolean {
        val logFilePath = provideGoLogFilePath()
        logger.log("Starting Go tunnel logger using local storage")
        if (!GoBackendWrapper.initLogger(logFilePath.toString())) {
            logger.log("[ERROR] service_logger_init result=failed failure_code=LOCAL_LOGGER_REJECTED")
            return false
        }
        logger.log("service_logger_init result=success state=ready")
        return true
    }
}

File: swift_module/CommonDI/LoggerManagerImpl.swift
import app
import DobbyVPNRuntime

public class LoggerManagerImpl: LoggerManager {
    private var logs = NativeModuleHolder.logsRepository

    public func doInitLogger() -> Bool {
        let logFilePath = LogsRepository_iosKt.provideGoLogFilePath().normalized().description()

        logs.writeLog(log: "Starting Go tunnel logger using owner-only local storage")
        guard DobbyvpnInitLogger(logFilePath) else {
            logs.writeLog(log: "[ERROR] service_logger_init result=failed failure_code=LOCAL_LOGGER_REJECTED")
            return false
        }
        logs.writeLog(log: "service_logger_init result=success state=ready")

        return true
    }
}
